### ISSUE TITLE
Making getting the current time faster

### ISSUE THREAD

**Issue by bitemyapp** [bitemyapp/blacktip#1]
This sucks:
https://github.com/bitemyapp/blacktip/blob/master/src/Database/Blacktip.hs#L45

Worse, it sucks in multiple ways.

First:

`(round . (*1000))`

`round` is slow-as. The `(*1000)` ideally wouldn't even be necessary.

Second:

`PSX.getPOSIXTime` originates in a syscall and goes through like 3 redundant transformations before I get anything out of it. This is slow and unnecessary! Ideally we'd want a millisecond-resolution with the same 64-bit representation in as little work as is possible.

Worth noting that getPOSIXTime doesn't guarantee monotonicity. I don't know how likely of a problem that is.

**Comment by andrewthad** [comment]
The [clock](http://hackage.haskell.org/package/clock-0.7.2/docs/System-Clock.html) package is probably a better fit for this than the `time` package.

**Comment by bitemyapp** [comment]
@andrewthad yeah somebody mentioned it had monotonic clock but there's still some trickiness involved:

1. I'd still want to cache it (millisecond resolution?) so I'm not pinging the kernel over and over

2. I need to peg the monotonic clock against wall clock so that I know what arithmetic to do to get the nominal epoch time for each id

**Comment by andrewthad** [comment]
Both good points. The caching problem seems like something [auto-update](http://hackage.haskell.org/package/auto-update-0.1.4/docs/Control-AutoUpdate.html) handles, but the way I've seen it used in yesod, it typically operates at the one-second resolution, not at the one-millisecond resolution. I don't really know exactly how much overhead it adds.

**Comment by bitemyapp** [comment]
I remember it from mucking around in Yesod, I need to get around to reviewing auto-update's code.

@andrewthad what's your interest in this repo?

**Comment by andrewthad** [comment]
I'm using elasticsearch. I like flake IDs, but I need to quit using autogenerated IDs.

**Comment by bitemyapp** [comment]
@andrewthad I edited your comment, is that about right? If so, I think this is a reasonable use-case. Are you planning to PR some last mile production readiness stuff?

**Comment by andrewthad** [comment]
Yeah, that's about what I meant. I'll try to put together a PR soon.

**Comment by bitemyapp** [comment]
@andrewthad Aight, Imma clean house a little then.
